src/main/java/org/ndgf/endit/FlushTask.java

```diff
@@ -89,9 +90,20 @@ public FlushTask(FlushRequest request, Path requestDir, Path outDir, String type
             checksumValue = checksum.getValue();           
         }
 
-        String s = String.format("{ \"file_size\": %d, \"time\": %d, \"storage_class\": %s, \"action\": %s, \"path\": %s, \"checksum_type\":%s, \"checksum_value\":%s }",
-    				size, System.currentTimeMillis() / 1000, "\"" + storageClass + "\"", "\"migrate\"" , "\"" + path + "\"", "\"" + checksumType + "\"", "\"" + checksumValue + "\"");
-    	Files.write(requestFile, s.getBytes(Charsets.UTF_8));
+
+    	JsonObject jsObj = new JsonObject();
+    	jsObj.addProperty("file_size", size);
+    	jsObj.addProperty("time", System.currentTimeMillis() / 1000);
+    	jsObj.addProperty("storage_class", storageClass);
+    	jsObj.addProperty("action", "migrate");
+    	jsObj.addProperty("path", path);
+    	jsObj.addProperty("checksumType", checksumType);
+    	jsObj.addProperty("checksumValue", checksumValue);  	
+    	
+    	if (jsObj.size() != 0 )
+           FileUtils.write(requestFile.toFile(), jsObj.toString(),  StandardCharsets.UTF_8);
+        else
```

Comment:
This is very unlikely event. You can ignore it, or thrown an RuntimeException

Reply:
yes, I was doubting whether to put it at all or not. Since you also think it is unnecessary, so I dropped it.